File: laboratorios/lab02/l-javascript/clases/Curso.js
import { Horario } from "./Horario.js";

export class Curso {
  #id
  #nombre
  #horario

  constructor(id, nombre, horario) {
    this.#id = id
    this.#nombre = nombre
    // Composicion: un curso tiene un horario
    this.#horario = horario instanceof Horario ? horario : null
  }

  get id() {
    return this.#id
  }
  set id(id) {
    this.#id = id
  }

  get nombre() {
    return this.#nombre
  }
  set nombre(nombre) {
    this.#nombre = nombre
  }

  get horario() {
    return this.#horario
  }
  set horario(horario) {
    if (horario instanceof Horario) {
      this.#horario = horario
    }
  }
}

File: laboratorios/lab02/l-java/src/main/java/com/tuorg/poo/clases/Curso.java
package com.tuorg.poo.clases;

public class Curso {
    private String id;
    private String nombre;
    // Composicion: Un curso tiene un horario
    private Horario horario;

    public Curso(String id, String nombre, Horario horario) {
        this.id = id;
        this.nombre = nombre;
        this.horario = horario;
    }

    public String getId() {
        return id;
    }

    public String getNombre() {
        return nombre;
    }

    public Horario getHorario() {
        return horario;
    }

    public void setId(String id) {
        this.id = id;
    }

    public void setNombre(String nombre) {
        this.nombre = nombre;
    }

    public void setHorario(Horario horario) {
        this.horario = horario;
    }
}
